File: android/src/main/java/com/capacitor/plugin/nativelog/NativeLogPlugin.java
package com.capacitor.plugin.nativelog;

import android.util.Log;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "NativeLog")
public class NativeLogPlugin extends Plugin {

    private NativeLog implementation = new NativeLog();

    @PluginMethod
    public void log(PluginCall call) {
        String tag = call.getString("tag", "IonicCapacitorApp");
        String message = call.getString("message");

        // Log.d(tag, "NativeLogPlugin.log");

        if (message != null) {
            Log.d(tag, message);
            call.resolve();
        } else {
            call.reject("Missing message");
        }
    }
}

File: ios/Plugin/NativeLogPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(NativeLogPlugin)
public class NativeLogPlugin: CAPPlugin {
    private let implementation = NativeLog()

    @objc func log(_ call: CAPPluginCall) {
        let tag = call.getString("tag") ?? "IonicCapacitorApp"
        guard let message = call.getString("message") else {
            call.reject("Missing message")
            return
        }
        implementation.log(tag: tag, message: message)
        call.resolve()
    }
}
